### PR TITLE
Add `query_duration_seconds` and `query_failures` metrics

### PR DESCRIPTION
**Pull request by sgrebnov**
Part of https://github.com/spiceai/spiceai/issues/1283

1. Added the following metrics
 - `query_duration_seconds`
 - `query_failures`
   - Don't currently include `accelerated` tag as part of tags dimension (will be added separately)
2. Updated query execution logic to handle failures and correctly report metrics / update query history.
3. Added the following fields to`runtime.query_history` (inspired by [snowflake query_history](https://docs.snowflake.com/en/sql-reference/account-usage/query_history))
- `execution_status` - 0 means success, -1 failure. Int8 selected to enable support for more granular statuses - `cancelled`, etc in the future
- `error_message`
4. `execution_time` now uses Float32

### CODE REVIEW

File: crates/runtime/src/datafusion/query.rs [hunk 1]
```diff
@@ -64,112 +83,173 @@ pub struct Query {
...
-            if let Some(execution_time) = self
-                .end_time
-                .and_then(|end_time| end_time.duration_since(self.start_time).ok())
-                .and_then(|duration| u64::try_from(duration.as_millis()).ok())
-            {
-                self.execution_time = Some(execution_time);
+            self.execution_time = Some(duration.as_secs_f32());
+        }
+
+        self.protocol.to_string();
```
Comment (phillipleblanc) [inline crates/runtime/src/datafusion/query.rs]:
```suggestion
```

File: crates/runtime/src/datafusion/query/builder.rs [hunk 2]
```diff
@@ -40,6 +42,7 @@ impl QueryBuilder {
             query_id: Uuid::new_v4(),
             nsql: None,
             restricted_sql_options: None,
+            protocol: Protocol::Http,
```
Comment (phillipleblanc) [inline crates/runtime/src/datafusion/query/builder.rs]: We shouldn't assume this is HTTP - this should be a required parameter.

**Review by phillipleblanc — DISMISSED** [review]

File: crates/runtime/src/datafusion/query/builder.rs (hunk 2 above)
Comment (sgrebnov) [reply crates/runtime/src/datafusion/query/builder.rs]: Updated, thank you!

File: crates/runtime/src/datafusion/query.rs (hunk 1 above)
Comment (sgrebnov) [reply crates/runtime/src/datafusion/query.rs]: Thank you

**Review by phillipleblanc — APPROVED** [review]

**Pull request merged**